core/src/test/java/com/linecorp/armeria/common/stream/FilteredStreamMessageTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.stream;
+
+import static com.linecorp.armeria.common.stream.StreamMessageTest.newPooledBuffer;
+import static com.linecorp.armeria.common.stream.SubscriptionOptionTest.subscriptionOptions;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.fail;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.annotation.Nullable;
+
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.unsafe.ByteBufHttpData;
+
+import io.netty.buffer.ByteBufHolder;
+
+class FilteredStreamMessageTest {
+
+    @Test
+    void withPooledObjects() {
+        boolean filterSupportsPooledObjects = true;
```

Comment:
I think it's better to either use `@ParameterizedTest` or split into four tests.

Reply:
Ah, yeah, will use `@ParameterizedTest`. Thanks!